src/plugins/plot/src/plot/MCTPlotController.js

```diff
@@ -534,6 +554,32 @@ define([
         this.cursorGuide = !this.cursorGuide;
     };
 
+    MCTPlotController.prototype.getXKeyOption = function (key) {
+        return this.$scope.xKeyOptions.find(option => option.key === key);
+    };
+
+    MCTPlotController.prototype.isEnabledXKeyToggle = function () {
+        const isSinglePlot = this.$scope.xKeyOptions && this.$scope.xKeyOptions.length > 1 && this.$scope.series.length === 1;
+        const isFrozen = this.config.xAxis.get('frozen');
+        const inRealTimeMode = this.config.openmct.time.clock();
+
+        return isSinglePlot && !isFrozen && !inRealTimeMode;
+    };
+
+    MCTPlotController.prototype.toggleXKeyOption = function (lastXKey, series) {
+        const selectedXKey = this.$scope.selectedXKeyOption.key;
+        const dataForSelectedXKey = series.data
+            ? series.data[0][selectedXKey]
+            : undefined;
+
+        if (dataForSelectedXKey !== undefined) {
+            this.config.xAxis.set('key', selectedXKey);
+        } else {
+            this.config.openmct.notifications.error('Cannot change x-axis view as no data exists for this view type.');
+            this.$scope.selectedXKeyOption.key = lastXKey;
```

Comment:
@shefalijoshi , this is what I was mentioning did not make sense to me. I was doing console logs in this `toggleXKeyOption` function of `this.$scope.xKeyOptions`, and seeing the previously selected "option" with its key changed to the new selected key. So for instance, where i would expect to see something like `[{key: 'utc', name: 'UTC', $$hashkey: 35}, {key: 'local', name: 'Local', $$hashkey: 36}]` I was seeing something like `[{key: 'local', name: 'UTC', $$hashkey: 35}, {key: 'local', name: 'Local', $$hashkey: 36}]`, where i was on UTC and switched to Local. I'm sure if I went down the path of looking at angular source code it would make sense.... but I was hoping there would be some better documentation.

The select that uses this is in _mct-plot.html_.
```html
 <select
    ng-show="plot.isEnabledXKeyToggle()"
    ng-model="selectedXKeyOption.key"
    ng-change="plot.toggleXKeyOption('{{selectedXKeyOption.key}}', series[0])"
    class="gl-plot-x-label__select local-controls--hidden"
    ng-options="option.key as option.name for option in xKeyOptions"
>
</select>
```